### ISSUE TITLE
`anonymous-formation` duplicates defects, if 2+ formations in the scope

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#483]
Here is the code:

```eo
# App.
[] > app
  "1st" > f
  "2nd" > t
  malloc.of
    64
    [m]
      QQ.io.stdout > @
        t
  malloc.of
    52
    [o]
      QQ.io.stdout > @
        f
```

Here, the lint should report only two defects:
* the usage of `t` inside `[m]`
* the usage of `f` inside `[o]`

However, we  are getting this:

```xml
<defect line="9" severity="warning">It is not recommended to have anonymous formation accessing undefined object "t" inside the formation</defect>
<defect line="14" severity="warning">It is not recommended to have anonymous formation accessing undefined object "f" inside the formation</defect>
<defect line="9" severity="warning">It is not recommended to have anonymous formation accessing undefined object "t" inside the formation</defect>
<defect line="14" severity="warning">It is not recommended to have anonymous formation accessing undefined object "f" inside the formation</defect>
```

Let's make the lint to not duplicate warnings for multiple formations

**Comment by h1alexbel** [comment]
@yegor256 looks like new bug

**Comment by yegor256** [comment]
@h1alexbel I agree, thanks for reporting!
